gratipay/models/participant/__init__.py

```diff
@@ -341,6 +341,14 @@ def clear_takes(self, cursor):
         """Leave all teams by zeroing all takes.
         """
         for team in self.get_teams():
+            if team.owner == self.username and team.get_take_for(self) == 0:
+
+                # Team owners don't necessarily have takes, and we also don't
+                # hard-require team owners to have a national identity on
+                # file. But we *do* require national identity before we will
+                # set a take--even if setting it to zero.
+
+                continue
```

Comment:
Another option, is to check whether the user is a member using the [`p.member_of(team)`](https://github.com/gratipay/gratipay.com/blob/master/gratipay/models/participant/__init__.py#L796) function. It'll blow up the number of queries, but taking over a participant is a rare event - I wouldn't worry about it right now.